kernel/nativelib.ml

```diff
@@ -183,11 +183,24 @@ let call_linker ?(fatal=true) ~prefix f upds =
      else if !Flags.debug then Feedback.msg_debug CErrors.(iprint exn));
   match upds with Some upds -> update_locations upds | _ -> ()
 
-let link_library ~prefix ~dirname ~basename =
+let link_library dirname libname =
+  let prefix = mod_uid_of_dirpath libname ^ "." in
+  let basename = Dynlink.adapt_filename (prefix ^ "cmo") in
   (* We try both [output_dir] and [.coq-native], unfortunately from
      [Require] we don't know if we are loading a library in the build
      dir or in the installed layout *)
   let install_location = dirname / dft_output_dir / basename in
   let build_location = dirname / !output_dir / basename in
   let f = if Sys.file_exists build_location then build_location else install_location in
   call_linker ~fatal:false ~prefix f None
+
+let delayed_link = ref []
```

Comment:
Does this mean that if I we have a document:
```coq
Definition a := 3.
Require foo.
```
and I evaluate to the `Require`, then remove it, the contents of this variable do become stale?

Reply:
Yes, but that is not different from the current behavior: You cannot unload a dynamically loaded module; you have to kill Coq entirely to do that. (And that is not a bug of Coq; that is a "feature" of OCaml.) This commit does not change that.

Reply:
Even if the current behavior is indeed tricky, this adds even more staleness, so I'd rather have that variable synchronized.

Reply:
This just does not make sense. Consider the following code:
```coq
Definition a := 3.
Require foo.
Eval native_compute in a.
```
and imagine you undo just the last command and redo it. This causes `delayed_link` to be restored and then fully handled again. As a consequence, Coq will try to dynamically load all the already loaded libraries a second time, which is bad. It is important that this variable is not synchronized.

Reply:
IIANM this is what happens with plugins already, so there is a check to avoid double-loading, but indeed, how to best do here is not so clear.